fairscale/optim/oss.py

```diff
@@ -107,6 +104,11 @@ def __init__(
         # Current default device is set by the parameters allocated to this rank
         self._device = list(self.per_device_params.keys())[0]
         self.buckets: Dict[torch.device, List[Bucket]] = {}
+
+        if self.world_size <= 8:
```

Comment:
you can use torch.cuda.get_device_count()?

Reply:
so ` if self.world_size == torch.cuda.get_device_count()` really ? yeah, makes sense